[18.0][MIG] l10n_es_intrastat_report: Migration to 18.0

LauraCForgeFlow opened the pull request:
Depends on:

- [x] #4033 

Also includes #4336 in a separate commit.

luc-demeyer committed:
add intrastat module

danielduqma committed:
[11.0][ADD] l10n_es_intrastat_product_declaration

Commit:
[MIG] l10n_es_intrastat_product: Migration to 12.0

pedrobaeza committed:
[MIG+IMP+FIX] l10n_es_intrastat_report: Finish migration to 12 + several imps/fixes

* Remove unneeded dependency to stock_picking_invoice_link.
* Updated HS Codes source file.
* Optimized compute/import methods.
* Remove useless/obsolete code, data and fields.
* Added Spanish translation.

pedrobaeza committed:
[FIX] l10n_es_intrastat_report: Allow invoices with no linked purchases

Previous code was forcing the existence of linked purchase lines which may be not
true for all. As we already have a fallback in declaration code, let's not force
this at location level.

joao-p-marques committed:
[IMP] l10n_es_intrastat_report: black, isort, prettier

joao-p-marques committed:
[MIG] l10n_es_intrastat_report: Migration to 13.0

TT27865

joao-p-marques committed:
[FIX] l10n_es_intrastat_report: Update multi-company rules

Missed step from the migration to v13
Update multi-company security domains and provide migration scripts

TT27865

joao-p-marques committed:
[MIG] l10n_es_intrastat_report: Migration to 14.0

pedrobaeza committed:
[FIX] l10n_es_intrastat_report: Fix tests

Due to upstream change, now invoice date is required.

dosipchuk committed:
[FIX] l10n_es_intrastat_report: additional check if company's country is set

Co-authored-by: Pedro M. Baeza <pedro.baeza@tecnativa.com>

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.1.0.1

victoralmau committed:
[IMP] l10n_es_intrastat_report: Update intrastat transaction data according to documentation.
TT30682

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.1.1.0

ValentinVinagre committed:
[IMP] l10n_es_intrastat_report: adaptacion para la nueva presentacion de 2022

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

victoralmau committed:
[IMP] l10n_es_intrastat_report: Add to notes when missing partner_vat and show error when missing product_origin_country_id in line.
TT29332

victoralmau committed:
[IMP] l10n_es_intrastat_report: Add to notes when missing partner_vat and show error when missing product_origin_country_id in line.
TT29332

victoralmau committed:
[FIX] l10n_es_intrastat_report: Change the link to the codes.
TT34295

victoralmau committed:
[FIX] l10n_es_intrastat_report: Change the link to the codes.
TT34295

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.2.0.1

OCA-git-bot committed:
l10n_es_intrastat_report 14.0.2.1.0

pedrobaeza committed:
[IMP] l10n_es_intrastat_report: 2022 NC codes

Updated from https://sede.agenciatributaria.gob.es/Sede/estadisticas/estadisticas-comercio-exterior/nomenclatura-combinada-ano.html

TT35871

pedrobaeza committed:
[FIX] l10n_es_intrastat_report: Test at post-install

Since 81aac30dd2278e43a43cf8fd6cef31e1e8c60f3f, all tests depending on
invoicing base test should be post-install.

flachica committed:
[MIG] l10n_es_intrastat_report 15.0

victoralmau committed:
[FIX] l10n_es_intrastat_report: Set the correct country code in csv file.

TT39888

OCA-git-bot committed:
l10n_es_intrastat_report 15.0.1.0.1

victoralmau committed:
[IMP] l10n_es_intrastat_report: Improve tests to test an invoice with delivery address in a country other than the country of contact.

TT40764

victoralmau committed:
[IMP] l10n_es_intrastat_report: 2023 NC codes

Updated from https://sede.agenciatributaria.gob.es/Sede/estadisticas/estadisticas-comercio-exterior/nomenclatura-combinada-ano.html

TT41641

victoralmau committed:
[FIX] l10n_es_intrastat_report: Search archived records when importing codes.

If any code is archived, we need to find it as well to avoid the duplicate code error.

TT41641

OCA-git-bot committed:
l10n_es_intrastat_report 15.0.2.0.0

OCA-git-bot committed:
l10n_es_intrastat_report 15.0.2.0.1

RabbitJon-S73 committed:
[MIG] l10n_es_intrastat_report: Migration to 16.0

Commit:
[UPD] Update l10n_es_intrastat_report.pot

victoralmau committed:
[IMP] l10n_es_intrastat_report: Set the fiscal position "Régimen Intracomunitario" with Intrastat flag

TT45420

victoralmau committed:
[FIX] l10n_es_intrastat_report: Change fp_intra_private to b2c (type of transaction)

TT45420

Commit:
[UPD] Update l10n_es_intrastat_report.pot

victoralmau committed:
[FIX] l10n_es_intrastat_report: Remove transactions, the data is already defined in intrastat_product.

Transaction 70 is eliminated, it does not exist (officially).

https://eur-lex.europa.eu/legal-content/ES/TXT/HTML/?uri=CELEX:32020R1197&from=EN#d1e32-7-1

company_id=False is not necessary in v16 because the field is no longer has a default

victoralmau committed:
[FIX] l10n_es_intrastat_report: Move methods to the correct class + Improve test

Forgotten changes in https://github.com/OCA/l10n-spain/pull/3111/commits/dc2555c6c53bcb3ff67ebc358c34dcfd6698ccb1 required due to change https://github.com/OCA/intrastat-extrastat/commit/7de2169b70d82e22e0c411c9a5a31c5fddc0419e

Commit:
[UPD] Update l10n_es_intrastat_report.pot

Commit:
[UPD] Update l10n_es_intrastat_report.pot

aritzolea committed:
[FIX] l10n_es_intrastat_report: Do not display missing VAT number error when fiscal position is B2C

aritzolea committed:
[FIX] l10n_es_intrastat_report: Adapt to changes on confirmation

Josep-s73 committed:
[IMP] l10n_es_intrastat_report: pre-commit auto fixes

Josep-s73 committed:
[17.0][MIG] l10n_es_intrastat_report

LauraCForgeFlow committed:
[IMP] l10n_es_intrastat_report: pre-commit auto-fixes

LauraCForgeFlow committed:
[MIG] l10n_es_intrastat_report: Migration to 18.0

acysos commented:
/ocabot migration l10n_es_intrastat_report

victoralmau reviewed (commented):
Aunque revisando el código parece ser correcto, ¿puedes hacer rebase para probarlo funcionalmente y revisar si sigue habiendo errores en los tests?

TT57210

victoralmau committed:
[FIX] l10n_es_intrastat_report: Define the appropriate fiscal positions with Intrastat

Since v17, ir.model.data records (external identifiers) are no longer created with
the localization module (l10n_es). Account is defined in all of them, so the
post_init_hook() method must be adjusted so that the appropriate fiscal positions
have the appropriate Intrastat data.

pedrobaeza commented:
Hay que incluir #4336

victoralmau commented:
Puedes añadir el commit en el lugar adecuado para respetar el histórico? (antes del commit de pre-commit).

<img width="1063" height="358" alt="ejemplo" src="https://github.com/user-attachments/assets/a26746dd-823a-47ac-b63a-385d53dee79e" />

victoralmau commented on l10n_es_intrastat_report/views/l10n_es_intrastat_product.xml:
```diff
@@ -31,7 +31,7 @@
         id="l10n_es_intrastat_product_computation_line_view_tree"
         model="ir.ui.view"
     >
-        <field name="name">intrastat.product.computation.line.tree</field>
```
No estoy seguro de que merezca la pena cambiar el `name` de las vistas (de tree a list), en cualquier caso, si se hace, habría que añadir un script de migración con rename_xmlids() para renombrarlas en pre-migration.

pedrobaeza replied:
Bueno, el problema sería cambiar el XML-ID, no el name.

victoralmau replied:
Ups, perdón, cierto, tienes razón.

LauraCForgeFlow commented:
¡Hecho! @victoralmau

victoralmau committed:
[FIX] l10n_es_intrastat_report: Define the appropriate data in post_init_hook

Define the appropriate data (intrastat+vat_required) in the fiscal positions
to be consistent with what is indicated in the _get_fiscal_position() method
of account.chart.template

victoralmau commented:
Puedes incluir también https://github.com/OCA/l10n-spain/pull/4347 ? (antes del commit de pre-commit).

pedrobaeza reviewed (approved):
/ocabot merge nobump
